src/main/java/org/example/service/UserService.java

```diff
@@ -42,13 +43,27 @@ public User saveUser(UserDTO userDTO) {
                 .populateBeanFromDTO(user, userDTO));
     }
 
-    public User updateUser(UserDTO userDTO, int id) {
-        User user = userRepository.findById(id).orElse(null);
+    public User saveAdmin(UserDTO userDTO) {
+        org.example.entity.User user = new org.example.entity.User();
+        user.setCreateTime(LocalDateTime.now());
         user.setUpdateTime(LocalDateTime.now());
+        user.setPassword(passwordEncoder.encode(userDTO.getPassword()));
+        user.setRole(roleRepository.findById(ADMIN_ROLE).orElse(null));
         return userRepository.save(transformerDTOUser
                 .populateBeanFromDTO(user, userDTO));
     }
 
+    public User updateUser(UserDTO userDTO, int id) {
+        User user = userRepository.findById(id).orElse(null);
```

Comment:
It's not correcr to use `orElse(null)` here, since we can get NPE
It's better to use `.get()` instead